ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,273 @@
+        CORTEX_LEFT_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_RIGHT_end_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
+
+        CORTEX_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
+        CEREBELLUM_start_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_LEFT').ciftilist);
+        CEREBELLUM_end_idx=max(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_RIGHT').ciftilist);
+        BRIAN_STEM_start_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'BRAIN_STEM').ciftilist);
+        BRIAN_STEM_end_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'BRAIN_STEM').ciftilist);
+        
+        row_names=cell(num_comps, 1);
+        probs=zeros(num_comps,1);
+        for k=1:num_comps
+            row_names{k}=[subj '_' fMRIName '_' num2str(k)];
+            probs(k,:)=probability(k);
+            OrigFeatures(k,:)=FeaturesFile(k,:);
+
+            NewFeatures(k,1)=Stats(k,1); %Percent explained variance
+            NewFeatures(k,2)=Stats(k,2); %Percent total variance
+            NewFeatures(k,3)=sum(Powerspectra(PowerspectraScale>0.25,k))./sum(Powerspectra((PowerspectraScale>0.005).*(PowerspectraScale<0.1)==1,k)); %Ratio of power spectrum less than 0.005 to BOLD range
+            NewFeatures(k,4)=sum(Powerspectra(PowerspectraScale<0.005,k))./sum(Powerspectra((PowerspectraScale>0.005).*(PowerspectraScale<0.1)==1,k)); %Ratio of power spectrum greater than 0.25 to BOLD range
+            NewFeatures(k,5)=sum(abs(sICAweighted(sICAnoisy,k)))./sum(abs(sICAweighted(:,k))); %Ratio of amplitude at noisy (top 12.5% variance) timepoints to all timepoints weighted by explained variance
+            NewFeatures(k,6)=sum(abs(sICA(sICAnoisy,k)))./sum(abs(sICA(:,k))); %Ratio of amplitude at noisy (top 12.5% variance) timepoints to all timepoints unweighted
+            NewFeatures(k,7)=sum(abs(CIFTI.cdata(:,k)))./sum(abs(Volume_data(:,k))); %CIFTI/Volume
+            NewFeatures(k,8)=sum(abs(CIFTI.cdata(CORTEX_start_idx:CORTEX_end_idx,k)))./sum(abs(Volume_data(:,k))); %Cerebral Cortex/Volume
+            NewFeatures(k,9)=sum(abs(CIFTI.cdata(CORTEX_end_idx:end,k)))./sum(abs(Volume_data(:,k))); %Subcortical/Volume
+            NewFeatures(k,10)=sum(abs(CIFTI.cdata(CEREBELLUM_start_idx:CEREBELLUM_end_idx,k)))./sum(abs(Volume_data(:,k))); %Cerebellum/Volume
+            NewFeatures(k,11)=sum(abs(CIFTI.cdata(BRIAN_STEM_start_idx:BRIAN_STEM_end_idx,k)))./sum(abs(Volume_data(:,k))); %Brainstem/Volume
+            NewFeatures(k,12)=sum(abs(CIFTI.cdata(setdiff(1:CIFTI_LENGTH,[CORTEX_start_idx:CEREBELLUM_end_idx BRIAN_STEM_start_idx:BRIAN_STEM_end_idx CEREBELLUM_start_idx:CEREBELLUM_end_idx]),k)))./sum(abs(Volume_data(:,k))); %Diencephalon/Volume
```

Comment:
Typo: `CORTEX_start_idx:CEREBELLUM_end_idx`, with CEREBELLUM again later

If you use lists like I suggest (rather than start/end), this `setdiff` becomes `setdiff(1:CIFTI_LENGTH, [CORTEX_idx_list, BRAIN_STEM_idx_list, CEREBELLUM_idx_list])`, which is simpler, more readable, and less error prone.

Reply:
Yes, I see.

Reply:
IMPORTANT: this "cerebellum" typo has probably excluded accumbens, amygdala, and caudate from this feature in previous runs!

Reply:
This typo is only included after this commit so won't affect the weights. 

This is what it used to be: `setdiff(1:91282,[1:29696+29716 60334+1:60334+1+3472 65289+1:83142]`

Reply:
Ah, I see.  So, the generated data is good, as long as those numbers were right (which is difficult to verify on review when they are hardcoded, even for the case of standard 91282).

Reply:
Technically, I think `60334+1:60334+1+3472` included one too many indices, so there is a voxel of left caudate included in your brain stem.  Otherwise, the hardcoded values look plausible.